usr/share/rear/lib/write-protect-functions.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+#
+# Functions to identify write-protected disks and partitions
+#
+
+function write_protected_candidate_device() {
+    local device="$1"
+    # prints the path of the block device, translating it if given as /sys/block/*.
+
+    if [[ "$device" == /sys/block/* ]]; then
+        device="$(get_device_name "$device")"
+    fi
+    [[ ! -b "$device" ]] && Error "Could not check '$1' ('$device') for write protection – not a block device"
+    echo "$device"
+}
+
+function is_write_protected_by_pt_uuid() {
+    local device="$(write_protected_candidate_device "$1")"
+    # returns 0 if the device's partition table UUID is in the list of write-protected UUIDs.
+
+    local partition_table_uuid="$(lsblk --output PTUUID --noheadings --nodeps "$device")"
```

Comment:
older `lsblk` do not support `PTUUID`
so some fallback code if partition_table_uuid is empty should be there.
In general the `lsblk` call may fail for whatever reason,
cf. "Try hard to care about possible errors" in
https://github.com/rear/rear/wiki/Coding-Style

Reply:
The fallback code is there: An empty partition table UUID would never match. The array's elements are separated by a single space and two consecutive spaces as in `*"  "*` do not match anywhere.